shared/common.py

```diff
@@ -27,12 +31,38 @@ def client(self, service_name: str):
         return self.session.client(service_name, region_name=self.region_name)
 
 
-class Resource(NamedTuple):
+class ResourceDigest(NamedTuple):
     id: str
-    name: str
     type: str
-    details: str
-    group: str
+
+
+class ResourceEdge(NamedTuple):
+    from_node: ResourceDigest
+    to_node: ResourceDigest
+    label: str = None
+
+
+class Resource(NamedTuple):
+    digest: ResourceDigest
+    name: str
+    details: str = ""
+    group: str = ""
+
+
+class ResourceProvider:
+    def __init__(self):
+        """
+        Base provider class that provides resources and relationships.
+
+        The class should be implemented to return resources of the same type
+        """
+        self.relations_found: List[ResourceEdge] = []
+
+    def get_resources(self) -> List[Resource]:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/Cloud-Architects/cloud-discovery/pullRequest?prid=5655786)